### ISSUE TITLE
Segfault in last patch

### ISSUE THREAD

**Issue by FjjDessoyCaraballo** [FjjDessoyCaraballo/minishell#78]
Commit c465c2761d327dd29928e019fe0eacde91322987 introduced segfault.

LOG:

```shell
READ of size 8 at 0x604000002258 thread T0
    #0 0x4cd044 in free_tokens /home/fdessoy-/projects/minishell/src/line_handler.c:36:18
    #1 0x4cb5a3 in free_gang /home/fdessoy-/projects/minishell/src/main.c:37:2
    #2 0x4cb6ab in main /home/fdessoy-/projects/minishell/src/main.c:68:3
    #3 0x7c428e429d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #4 0x7c428e429e3f in __libc_start_main csu/../csu/libc-start.c:392:3
    #5 0x41f3d4 in _start (/home/fdessoy-/projects/minishell/minishell+0x41f3d4)

0x604000002258 is located 8 bytes inside of 48-byte region [0x604000002250,0x604000002280)
freed by thread T0 here:
    #0 0x49a032 in free (/home/fdessoy-/projects/minishell/minishell+0x49a032)
    #1 0x4cc4f4 in free_null /home/fdessoy-/projects/minishell/src/utils2.c:17:2
    #2 0x4cd155 in free_tokens /home/fdessoy-/projects/minishell/src/line_handler.c:44:3
    #3 0x4cb5a3 in free_gang /home/fdessoy-/projects/minishell/src/main.c:37:2
    #4 0x4cb69d in main /home/fdessoy-/projects/minishell/src/main.c:66:4
    #5 0x7c428e429d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16

previously allocated by thread T0 here:
    #0 0x49a29d in malloc (/home/fdessoy-/projects/minishell/minishell+0x49a29d)
    #1 0x4e08ce in ft_calloc /home/fdessoy-/projects/minishell/libft/libft/ft_calloc.c:23:8
    #2 0x4d7176 in init_token /home/fdessoy-/projects/minishell/src/parse/init_token.c:19:21
    #3 0x4d5412 in initialize_tokens /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:81:27
    #4 0x4d5517 in line_tokenization /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:91:24
    #5 0x4cd64d in sniff_line /home/fdessoy-/projects/minishell/src/line_handler.c:89:9
    #6 0x4cb641 in main /home/fdessoy-/projects/minishell/src/main.c:56:13
    #7 0x7c428e429d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
```

**Comment by FjjDessoyCaraballo** [comment]
```shell
==134267==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 16 byte(s) in 5 object(s) allocated from:
    #0 0x49a29d in malloc (/home/fdessoy-/projects/minishell/minishell+0x49a29d)
    #1 0x4e0331 in remove_quotes /home/fdessoy-/projects/minishell/src/parse/ft_strtok_utils.c:154:29
    #2 0x4d646e in remove_quotes_and_skip_delimiters /home/fdessoy-/projects/minishell/src/parse/ft_strtok.c:45:20
    #3 0x4d70ff in ft_strtok /home/fdessoy-/projects/minishell/src/parse/ft_strtok.c:162:5
    #4 0x4d5cc8 in line_tokenization /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:108:25
    #5 0x4cd64d in sniff_line /home/fdessoy-/projects/minishell/src/line_handler.c:89:9
    #6 0x4cb641 in main /home/fdessoy-/projects/minishell/src/main.c:56:13
    #7 0x719365229d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16

Direct leak of 12 byte(s) in 4 object(s) allocated from:
    #0 0x49a29d in malloc (/home/fdessoy-/projects/minishell/minishell+0x49a29d)
    #1 0x4e0b5b in ft_strdup /home/fdessoy-/projects/minishell/libft/libft/ft_strdup.c:25:8
    #2 0x4d5972 in line_tokenization /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:100:25
    #3 0x4cd64d in sniff_line /home/fdessoy-/projects/minishell/src/line_handler.c:89:9
    #4 0x4cb641 in main /home/fdessoy-/projects/minishell/src/main.c:56:13
    #5 0x719365229d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16

Direct leak of 2 byte(s) in 1 object(s) allocated from:
    #0 0x49a29d in malloc (/home/fdessoy-/projects/minishell/minishell+0x49a29d)
    #1 0x4e0331 in remove_quotes /home/fdessoy-/projects/minishell/src/parse/ft_strtok_utils.c:154:29
    #2 0x4d646e in remove_quotes_and_skip_delimiters /home/fdessoy-/projects/minishell/src/parse/ft_strtok.c:45:20
    #3 0x4d70ff in ft_strtok /home/fdessoy-/projects/minishell/src/parse/ft_strtok.c:162:5
    #4 0x4d564c in line_tokenization /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:92:21
    #5 0x4cd64d in sniff_line /home/fdessoy-/projects/minishell/src/line_handler.c:89:9
    #6 0x4cb641 in main /home/fdessoy-/projects/minishell/src/main.c:56:13
    #7 0x719365229d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16

SUMMARY: AddressSanitizer: 30 byte(s) leaked in 10 allocation(s).
```

**Comment by FjjDessoyCaraballo** [comment]
CURRENT STATE:

```shell
==134808==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 7 byte(s) in 2 object(s) allocated from:
    #0 0x49a29d in malloc (/home/fdessoy-/projects/minishell/minishell+0x49a29d)
    #1 0x4e0b0b in ft_strdup /home/fdessoy-/projects/minishell/libft/libft/ft_strdup.c:25:8
    #2 0x4d5922 in line_tokenization /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:100:25
    #3 0x4cd5fd in sniff_line /home/fdessoy-/projects/minishell/src/line_handler.c:86:9
    #4 0x4cb641 in main /home/fdessoy-/projects/minishell/src/main.c:56:13
    #5 0x7d8e70029d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16

Direct leak of 3 byte(s) in 1 object(s) allocated from:
    #0 0x49a29d in malloc (/home/fdessoy-/projects/minishell/minishell+0x49a29d)
    #1 0x4e02e1 in remove_quotes /home/fdessoy-/projects/minishell/src/parse/ft_strtok_utils.c:154:29
    #2 0x4d641e in remove_quotes_and_skip_delimiters /home/fdessoy-/projects/minishell/src/parse/ft_strtok.c:45:20
    #3 0x4d70af in ft_strtok /home/fdessoy-/projects/minishell/src/parse/ft_strtok.c:162:5
    #4 0x4d55fc in line_tokenization /home/fdessoy-/projects/minishell/src/parse/tokenizer.c:92:21
    #5 0x4cd5fd in sniff_line /home/fdessoy-/projects/minishell/src/line_handler.c:86:9
    #6 0x4cb641 in main /home/fdessoy-/projects/minishell/src/main.c:56:13
    #7 0x7d8e70029d8f in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16

SUMMARY: AddressSanitizer: 10 byte(s) leaked in 3 allocation(s).
```
 
 We need to handle these leaks.

**Comment by FjjDessoyCaraballo** [comment]
Segfaults solved already in commit  b6c311f337484d0af7f8f786a344f2200d9d63ad
